src/Typechecker/Monad.hs

```diff
@@ -135,6 +138,12 @@ data TypeError = Mismatch Type Type (Expr SourcePos) SourcePos     -- ^ Couldn't
                | Unknown String SourcePos             -- ^ Errors that "shouldn't happen"
                | BadOp Op Type Type (Expr SourcePos) SourcePos    -- ^ Can't perform a primitive operation
                | OutOfBounds Pos Pos SourcePos
+               deriving (Eq)
+
+instance ToJSON TypeError where
+  toJSON (Mismatch t1 t2 _ src) = object ["errtype" .= String "mismatch", "left" .= t1, "right" .= t2, "line" .= sourceLine src, "col" .= sourceColumn src]
+  toJSON (NotBound n src) = object ["errtype" .= String "notBound", "name" .= n, "line" .= sourceLine src, "col" .= sourceColumn src]
+  toJSON _ = object ["errtype" .= String "ERROR TODO"]
```

Comment:
Okay, alright this makes a bit more sense now with the replacement you made above. I want to double check the resulting error tag however. As all the existing responses go:
```json
{"tag": "SpielThing", "contents": "blah"}
```